### ISSUE TITLE
Ordered stdout - [closed]

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#1123]
In GitLab by @methane on Dec 27, 2014, 10:19

_Merges ordered-stdout -> master_

Fixes #14

Cons: stdout is not written realtime.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Dec 27, 2014, 10:21

Build triggered.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Dec 27, 2014, 10:23

Build finished.  Tests PASSED.
Build results available at: [http://ci.sigmavir.us/job/flake8/28/](http://ci.sigmavir.us/job/flake8/28/)

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Dec 27, 2014, 10:29

Yeah this is why I marked #14 as "won't fix". I'm not confident we don't want real-time printing. At least I'm fairly certain people will complain about the behaviour change that results from this. I'm okay with non-deterministic file-ordering personally.

**Comment by asottile** [comment]
In GitLab by @RonnyPfannschmidt on Feb 22, 2015, 05:55

order changes utterly suck - flak8 regulary hides the last item of file x before a wall of other items from file y when cleaning up a project
and when things are clean it shouldnt matter as well

**Comment by asottile** [comment]
In GitLab by @msabramo on Mar 5, 2015, 08:54

What about making it an option?

```
$ flake8 --help
Usage: flake8 [options] input ...

Options:
...
  --ordered-stdout           prints errors and warnings in a deterministic order
                             (disables real-time output)
...
```

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Mar 7, 2015, 18:28

So I have an idea for how to handle this but it won't be ready in time for 2.4.0. I'm adding this for 2.5.0.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Apr 18, 2015, 18:23

:white_check_mark: Jenkins Build Success

Results available at: [Jenkins](http://ci.sigmavir.us/job/flake8/86/)

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Dec 17, 2015, 07:39

mentioned in merge request !47

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Jun 15, 2016, 07:27

This will be fixed in flake8 3.0. We'll have more control over reporting and will be able to order our stdout consistently with little additional overhead.
